docs/service-release-checklist.md

```diff
@@ -0,0 +1,104 @@
+# Service readiness checklist
+
+This document aims to provide a checklist for determining whether service is mature enough for general availability.
+
+## Checklist
+
+Legend: `M` - Mandatory, `R` - Recommended, `O` - Optional
+
+### Documentation
+
+- `R`: README.md in the repository:
```

Comment:
In my opinion `README.md` should be mandatory. There's nothing worse than landing on a repository without one. I'm on the fence about the `Architectural diagram for the service` too, as having it would greatly improve the quality of the document (given it's in the README).